python/cudf/cudf/core/reshape.py

```diff
@@ -353,35 +388,65 @@ def concat(objs, axis=0, join="outer", ignore_index=False, sort=None):
             objs = _align_objs(objs, how=join, sort=sort)
             df.index = objs[0].index
 
-        for o in objs:
-            for name, col in o._data.items():
-                if name in df._data:
-                    raise NotImplementedError(
-                        f"A Column with duplicate name found: {name}, cuDF "
-                        f"doesn't support having multiple columns with "
-                        f"same names yet."
-                    )
-                if empty_inner:
-                    # if join is inner and it contains an empty df
-                    # we return an empty df, hence creating an empty
-                    # column with dtype metadata retained.
-                    df[name] = cudf.core.column.column_empty_like(
-                        col, newsize=0
-                    )
-                else:
-                    df[name] = col
+        only_series = typs == {cudf.Series}
```

Comment:
This is another case where we should be checking `isinstance` rather than equality (which is what the set check will use). I know this suggestion conflicts with one that @wence- posed previously, but I'm not sure what else to recommend.

Is there some way to refactor this to remove `typs` altogether?
```suggestion
        only_series = len(typs) == 1 and all(isinstance(typ, cudf.Series) for typ in typs)
```

Reply:
i'm having trouble finding another class that inherits from `cudf.Series`, and it's unclear to me why we need to make sure the types are the same -- it seems merely checking they're all instances of `cudf.Series` suffices.

Reply:
It may not be a subclass in this repository.

cuSpatial, for instance, has a `GeoSeries` that inherits from `cudf.Series`: https://github.com/rapidsai/cuspatial/blob/37f35f8f57d877523f7b942406d3a98e5937424c/python/cuspatial/cuspatial/core/geoseries.py#L54

Reply:
that's fair enough, but do we need to ensure they're all the same type? i don't see a strong reason based on the code we're touching

Reply:
I think so, yes.

Reply:
> I know this suggestion conflicts with one that @wence- posed previously, but I'm not sure what else to recommend.

I had forgotten about subclasses, so ignore me.